R/weighted_mean_median_sd_mad.R

```diff
@@ -25,6 +25,13 @@ weighted_mean <- function(x, weights = NULL, verbose = TRUE, ...) {
     return(mean(x, na.rm = TRUE))
   }
 
+  # remove missings
+  x[is.na(weights)] <- NA
+  weights[is.na(x)] <- NA
+
+  weights <- stats::na.omit(weights)
+  x <- stats::na.omit(x)
```

Comment:
We are using this same block of code twice. Maybe turn it into a small utility function?